### PR TITLE
CHI-991: Migrated all service tests from auth bypass to mock auth server. 

### PR DESCRIPTION
**Pull request by stephenhand**
Primary Reviewer @murilovmachado 

## Description

* Migrated all service tests from auth bypass to mock auth server being used by permissions test
* Removed auth bypass from codebase
* Removed auth bypass related environment variables from npm scripts and workflows
* Fixed issue where references to `case.test.ts` in other test files, which caused the tests to also be loaded - but with strange set up and teardown issues

### Checklist
- [X] Corresponding issue has been opened
- [X] New tests added

### Related Issues
Fixes CHI-991

### Verification steps

Service & E2E tests will verify

### CODE REVIEW

File: hrm-service/service-tests/case-audits.test.js [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
+  authTokenLookup: () => 'picernic basket',
+}).listen();
 const request = supertest.agent(server);
 
 const { case1, case2, accountSid } = mocks;
 
 const headers = {
   'Content-Type': 'application/json',
-  Authorization: `Basic ${Buffer.from(process.env.API_KEY).toString('base64')}`,
+  Authorization: `Bearer bearing a bear (rawr)`,
```
Comment (GPaoloni) [inline hrm-service/service-tests/case-audits.test.js:19]: :rofl:

File: hrm-service/service-tests/case-audits.test.js [hunk 2]
```diff
@@ -27,14 +31,18 @@ const caseAuditsQuery = {
 };
 
 beforeAll(async () => {
+  await proxiedEndpoints.start();
+  await proxiedEndpoints.mockSuccessfulTwilioAuthentication(workerSid);
```
Comment (GPaoloni) [inline hrm-service/service-tests/case-audits.test.js:35]: Would be possible to merge this with the server start, and use some config parameters? Just asking, no need to change anything here!

File: hrm-service/src/app.ts [hunk 3]
```diff
@@ -84,20 +83,6 @@ export function createService({
...
-
-        if (isTestSecretValid) {
-          const testUser = req.headers['test-user'] || 'worker-sid';
-          req.user = new User(testUser, []);
-          return next();
-        }
-
-        console.log('API Key authentication failed');
-      }
-
```
Comment (GPaoloni) [inline hrm-service/src/app.ts:100]: :raised_hands:

**Review by GPaoloni — APPROVED** [review]
Looks good to me! I think it will also be worth adding here (or somewhere) the messages you sent via Slack describing why importing the `workerSid` was such a big deal, for future reference.

File: hrm-service/service-tests/case-audits.test.js (hunk 2 above)
Comment (stephenhand) [reply hrm-service/service-tests/case-audits.test.js:35]: We can do - I was thinking we might be mocking out other endpoints in other tests in the future and we will want to pick and choose which endpoints we mock & how, separately to starting the server

**Pull request merged**